indico/modules/events/contributions/forms.py

```diff
@@ -46,8 +51,21 @@ class ContributionForm(IndicoForm):
 
     def __init__(self, *args, **kwargs):
         self.event = kwargs.pop('event')
+        self.contrib = kwargs.pop('contrib', None)
+        self.conf = self.event.as_legacy
+        self.timezone = self.conf.getTimezone()
         super(ContributionForm, self).__init__(*args, **kwargs)
         self.type.query = self.event.contribution_types
+        if self.contrib is None or not self.contrib.is_scheduled:
+            del self.start_date
+
+    def validate_duration(self, field):
+        duration = field.data
+        if duration > timedelta(days=1):
+            raise ValidationError(_('Duration can not span more than one day'))
+        start_dt = self.start_date
```

Comment:
`start_date_field`? Otherwise one first thinks that this contains an actual date